### ISSUE TITLE
Absinthe allows to use objects (non input objects) as valid fields arguments

### ISSUE THREAD

**Issue by virtual-light** [absinthe-graphql/absinthe#806]
### Environment
```
Elixir 1.9.0 (compiled with Erlang/OTP 22)
```
```
absinthe 1.4.16 (Hex package) (mix)
absinthe_relay 1.4.6 (Hex package) (mix)
absinthe_plug 1.4.7 (Hex package) (mix)
```

### Expected behavior

According to [current GQL specification](https://graphql.github.io/graphql-spec/June2018) paragraph [3.4.2](https://graphql.github.io/graphql-spec/June2018/#sec-Input-and-Output-Types) and [working draft of new GQL specification](https://graphql.github.io/graphql-spec/draft) paragraph [3.4.2](https://graphql.github.io/graphql-spec/draft/#sec-Input-and-Output-Types).

>Input Object types can only be used as input types. Object, Interface, and Union types **can only be used as output types**.

That means, that [object](https://hexdocs.pm/absinthe/Absinthe.Schema.Notation.html#object/3) can't be used as a field argument type. 
Ideally absinthe should detect such definitions at schema verification stage and treat it as specification error. Otherwise absinthe should throw error at least when trying to resolve field with such argument.

### Actual behavior
Currently absinthe allows to use Object as a type of field argument. 

Biggest part of popular GQL clients fails when trying to inspect the schema that has fields arguments with Object type.

**Comment by maartenvanvliet** [comment]
Can you reproduce this with the `1.5` branch?

If I do this on `master`
```
  object :user do
    field :name, :string
  end
  input_object :contact_input do
    field :user, :user
  end
```

I get an error 
```
:user is not a valid type for field :user because
:user is an ObjectTypeDefinition, and this field is part of an InputObjectTypeDefinition.

Only input types may be used as inputs. Input types may not be used as output types
```

**Comment by benwilson512** [comment]
Yeah this is fixed in the 1.5 releases, it will not be back ported to 1.4. 1.4 uses an older schema construction method and it isn't easy to make changes to at this point.
